libretro/libretro.c

```diff
@@ -1731,6 +1843,20 @@ static void update_variables(bool startup)
     }
 #endif // HAVE_THR_AL
 
+    var.key = CORE_NAME "-pak-swap-delay";
+    var.value = NULL;
+    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
+    {
+       if (!strcmp(var.value, "no delay"))
+         swap_pak_delay = 0;
+       else if (!strcmp(var.value, "1s hold"))
+         swap_pak_delay = 60;
+       else if (!strcmp(var.value, "2s hold"))
+         swap_pak_delay = 120;
+       else if (!strcmp(var.value, "3s hold"))
+         swap_pak_delay = 180;
```

Comment:
1. The heck is that unit? Frames?
2. If its Frames/VI, then this is not correct depending on Buffer Swap mode for GlideN & PAL/NTSC configurations
3. atoi or sscanf, prefer not to have val and string mixed tho

Reply:
Yeah it was supposed to be frames, I assumed 60fps and completely forgot about PAL 😓 
A bit annoying since `ROM_PARAMS` isn't initialized yet on the first `update_variables()` pass so it returns 0 (NTSC) as region even with a PAL game...

Would that be OK as a workaround (with libretro_core_options.h values adjusted to just "0/1/2/3")?
```cpp
    var.key = CORE_NAME "-pak-swap-delay";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       // If we're on the 1st pass we don't know the region of the game yet,
       // so we just keep the seconds for now, and we'll multiply by FPS
       // just a bit later in retro_load_game()
       swap_pak_delay = atoi(var.value);
       if (!startup)
          swap_pak_delay *= retro_get_region() == RETRO_REGION_PAL ? 50 : 60;
    }
```
then in retro_load_game() after we got the ROM infos:
```cpp
    // Now that we now the region, we can adjust swap_pak_delay
    swap_pak_delay *= retro_get_region() == RETRO_REGION_PAL ? 50 : 60;
```

As for the buffer swap I'm not sure how to handle that tbh 🤔 

A bit of a bummer, looks more complicated than I expected for my *very* limited C/C++ skills :(

Reply:
Changes pushed, AFAICT it works fine with the correct hold duration for NTSC/PAL now.

Still not sure to understand your "buffer swap" comment, doesn't retro_run run 60 (or 50 for PAL) times per second no matter what? Because basically the only thing I'm doing is increasing the frame count for the duration of the button being held once per retro_run.

Reply:
Different approach using `clock()` from time.h: https://github.com/bslenul/mupen64plus-libretro-nx/commit/b2d9c20ace3e8bf1bd28ce9d5a2419e60253686c#diff-0033c4d206c0f5ecb21459b1c42f39055aacfc7468b9729527c5dd94489c58aa

Otherwise I could simply drop the hold button feature and make the swap on button press only. I initially added the hold options to avoid misclicks, but if you think it's not worth I don't mind removing it.

Reply:
I will think of smth